libs/shared/shared_common.c

```diff
@@ -31,9 +31,8 @@ uint16_t _supported_ebpf_extension_version[] = {
     EBPF_PROGRAM_SECTION_INFORMATION_CURRENT_VERSION,
 };
 
-#define EBPF_ATTACH_PROVIDER_DATA_SIZE_0 \
-    EBPF_OFFSET_OF(ebpf_attach_provider_data_t, link_type) + sizeof(enum bpf_link_type)
-size_t _ebpf_attach_provider_data_supported_size[] = {EBPF_ATTACH_PROVIDER_DATA_SIZE_0};
+#define EBPF_ATTACH_PROVIDER_DATA_SIZE_1 EBPF_SIZE_INCLUDING_FIELD(ebpf_attach_provider_data_t, link_type)
```

Comment:
since this is a breaking change, we may as well not update the DATA_SIZE definition macro.

Reply:
The meaning of the fields has changed, hence the bump from v0 to v1.